### ISSUE TITLE
[Feature Request] Support Larastan v2

### ISSUE THREAD

**Issue by fylzero** [enlightn/enlightn#104]
**Is your feature request related to a problem? Please describe.**

Receiving errors when trying to install Enlightn on a project that has Larastan v2 installed.

As I was trying to figure this out I opened 2 issues that can be referenced here:
https://github.com/enlightn/enlightn/issues/103
https://github.com/nunomaduro/larastan/issues/1165

**Describe the solution you'd like**

Would be nice to have updated support/a new verision of Enlightn for Larastan v2.

**Describe alternatives you've considered**

Not sure if there are any other solutions.

**Additional context**

**Comment by potsky** [comment]
And given that Laravel 9 only supports Larastan v>2.0, I cannot use enlightnpro on Laravel 9 :-(

**Comment by potsky** [comment]
@paras-malhotra any chance to support larastan v2 ?

**Comment by Fadelwahed** [comment]
Hey 

we have a **PRO** account and we can't use it with laravel 9 can you please give us any details

**Comment by paras-malhotra** [comment]
@Fadelwahed can you post the error here? I can help you resolve the issue.

**Comment by potsky** [comment]
> @Fadelwahed can you post the error here? I can help you resolve the issue.

The error is that we cannot install enlightnpro via composer because of the dependancy to PHPStan V1 for enlightnpro and PHPStan v2 for Laravel 9...

**Comment by paras-malhotra** [comment]
@potsky, I am unable to understand the conflict.

- Enlightn Pro has no direct dependency on PHPStan or Larastan - it depends on Enlightn
- Enlightn has dependencies on PHPStan < 1.2 and Larastan ^1.0.0
- Larastan ^1.0.0 supports Laravel ^9.0

That's why I don't see any conflict. Perhaps there are some other dependencies in your project `composer.json` or I may be missing something?

**Comment by potsky** [comment]
> @potsky, I am unable to understand the conflict.
> 
> * Enlightn Pro has no direct dependency on PHPStan or Larastan - it depends on Enlightn
> * Enlightn has dependencies on PHPStan < 1.2 and Larastan ^1.0.0
> * Larastan ^1.0.0 supports Laravel ^9.0
> 
> That's why I don't see any conflict. Perhaps there are some other dependencies in your project `composer.json` or I may be missing something?
> 
> EDIT: There are also tests configured for Laravel 9 [here](https://github.com/enlightn/enlightn/blob/master/.github/workflows/tests.yaml#L12) which are working fine, so there shouldn't actually be a conflict.

If you use Laravel 9, you must use Larastan 2.x, there are a looooooooot of new tests specific to L9. In the Larastan documentation : _If you are using a Laravel version older than 9.x, please refer to [Larastan v1.x]._

So if you specify `"nunomaduro/larastan": "^2.1",` in you `composer.json`, it will crash. We just would like that enlightn this requirement.

**Comment by bsaqqa** [comment]
> @potsky, I am unable to understand the conflict.
> 
> * Enlightn Pro has no direct dependency on PHPStan or Larastan - it depends on Enlightn
> * Enlightn has dependencies on PHPStan < 1.2 and Larastan ^1.0.0
> * Larastan ^1.0.0 supports Laravel ^9.0
> 
> That's why I don't see any conflict. Perhaps there are some other dependencies in your project `composer.json` or I may be missing something?
> 
> EDIT: There are also tests configured for Laravel 9 [here](https://github.com/enlightn/enlightn/blob/master/.github/workflows/tests.yaml#L12) which are working fine, so there shouldn't actually be a conflict.

@paras-malhotra 

But If we need to use Laravel 9.x without directly using Larastan. So we need to run our Enlightn Pro package on our project now. And may in the future when you upgrade Enlightn to support Larastan 2.1, So we are updating Enlightn Pro to the last version. 

Now we need just what is the instructions to run Enlightn Pro on our Laravel "9.13" app.

**Comment by paras-malhotra** [comment]
> So if you specify "nunomaduro/larastan": "^2.1", in you composer.json, it will crash. We just would like that enlightn this requirement.

Sure, I understand. We will be working on supporting Larastan 2.x. In the meantime, if you are using Laravel 9x with Enlightn, please use Larastan 1.x. There will not be any conflict - of course, there will be some missing Larastan features introduced in 2x but the composer install will work.

**Comment by paras-malhotra** [comment]
> Now we need just what is the instructions to run Enlightn Pro on our Laravel "9.13" app.
@bsaqqa you can refer the installation instructions in our docs [here](https://www.laravel-enlightn.com/docs/getting-started/installation.html)

**Comment by paras-malhotra** [comment]
Larastan 2.x is now supported in Enlightn v2.0.0 via #117.

We are now working on adding the same support for Enlightn Pro. I'll update this thread once the upgrade is done!

**Comment by paras-malhotra** [comment]
Larastan 2.x is now also supported for Enlightn Pro in v2.0.0! 🎉

**Comment by potsky** [comment]
Thank you @paras-malhotra, it works and it's good to see this again!

<img width="357" alt="Screenshot 2022-06-12 - 08-09-37@2x" src="https://user-images.githubusercontent.com/408237/173219452-3fd94c3c-05a8-4a02-8944-f45a67d97e05.png">

**Comment by fylzero** [comment]
@paras-malhotra Still receiving errors when trying to install this package with Larastan 2.x

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires enlightn/enlightn ^2.0 -> satisfiable by enlightn/enlightn[v2.0.0, 2.x-dev].
    - enlightn/enlightn[v2.0.0, ..., 2.x-dev] require nunomaduro/larastan ^2.0 <= 2.1.6 -> found nunomaduro/larastan[dev-master, 2.0.0, ..., v2.1.6] but the package is fixed to v2.1.11 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    ```

**Comment by potsky** [comment]
> @paras-malhotra Still receiving errors when trying to install this package with Larastan 2.x
> 
> ```
> Your requirements could not be resolved to an installable set of packages.
> 
>   Problem 1
>     - Root composer.json requires enlightn/enlightn ^2.0 -> satisfiable by enlightn/enlightn[v2.0.0, 2.x-dev].
>     - enlightn/enlightn[v2.0.0, ..., 2.x-dev] require nunomaduro/larastan ^2.0 <= 2.1.6 -> found nunomaduro/larastan[dev-master, 2.0.0, ..., v2.1.6] but the package is fixed to v2.1.11 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
> ```

You need to "unfix" your larastan version in your `composer.json` like this for example:

```
"nunomaduro/larastan": "^2.1",
```

For the moment, enlightn only support `larastan<=2.1.6` so if you have set it to `2.1.11` in your `composer.json`, composer cannot find a suitable version for your requirement and for enlightn.

I have asked to @paras-malhotra in the [PR](https://github.com/enlightn/enlightn/pull/117#issuecomment-1153080514) why enlightn is locked to `larastan<=2.1.6`...

**Comment by paras-malhotra** [comment]
@potsky it isn't compatible with `2.1.11` because of this issue: https://github.com/nunomaduro/larastan/pull/1260#issuecomment-1152944828

The way Larastan is written, we cannot override the Request stub file in 2.1.6+

**Comment by paras-malhotra** [comment]
Quick update: Larastan 2.x is now fully supported (including the latest version) in Enlightn v2.1.0 via #118
